data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/DataType.java

```diff
@@ -75,4 +75,22 @@ public String getTypeName() {
     static DataType fromTypeName(final String option) {
         return TYPES_MAP.get(option);
     }
+
+    public static boolean isSameType(final Object object, final String option) {
+        DataType type = fromTypeName(option);
+        if (type == null)
+            throw new IllegalArgumentException("Unknown DataType");
+        switch (type) {
```

Comment:
we should add a case condition for BigDecimal.

Reply:
After discussion, decided not to support these two for now as double handles float and all BigDecimals are actually strings.